PostFreeSurfer/scripts/CreateMyelinMaps.sh

```diff
@@ -242,60 +241,128 @@ for STRING in "$AtlasSpaceFolder"/"$NativeFolder"@native@roi "$AtlasSpaceFolder"
+				-right-spheres ${SphereFolder}/${Subject}.R.sphere.${RefResMesh}k_fs_LR.surf.gii ${AtlasSpaceFolder}/fsaverage_LR32k/${Subject}.R.sphere.32k_fs_LR.surf.gii \
+				-right-area-surfs ${T1wSurfFolder}/${Subject}.R.midthickness.${RefResMesh}k_fs_LR.surf.gii ${StudyFolder}/${Subject}/T1w/fsaverage_LR32k/${Subject}.R.midthickness.32k_fs_LR.surf.gii
 	fi
-	${CARET7DIR}/wb_command -cifti-separate ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.${MyelinMap}_BC.native.dscalar.nii COLUMN \
-		-metric CORTEX_LEFT ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.L.${MyelinMap}_BC.native.func.gii \
-		-metric CORTEX_RIGHT ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.R.${MyelinMap}_BC.native.func.gii
-done
-
-# create cifti and gift MyelinMap in the low res mesh spaces
-for LowResMesh in "${LowResMeshesArray[@]}" ; do
+	BiasFieldComputed=false
+	#Reduce memory usage by smoothing on downsampled mesh (match the gifti version by using the first lowresmesh)
+	LowResMesh="${LowResMeshesArray[0]}"
+	# myelin map only loop
+	for MyelinMap in MyelinMap SmoothedMyelinMap ; do
+		if [ "$BiasFieldComputed" = false ]; then
+			# ----- Begin moved statements -----
+			# Recompute Myelin Map Bias Field Based on Better Registration
+			log_Msg "Recompute Myelin Map Bias Field Based on Better Registration"
+			
+			# Myelin Map BC using low res
+			"$HCPPIPEDIR"/global/scripts/MyelinMap_BC.sh \
+				--study-folder="$StudyFolder" \
+				--subject="$Subject" \
+				--registration-name="MSMSulc" \
+				--use-ind-mean="$UseIndMean" \
+				--low-res-mesh="$LowResMesh" \
+				--myelin-target-file="$MyelinTargetFile" \
+				--map="$MyelinMap"
+			# ----- End moved statements -----
+			# bias field is computed in the module MyelinMap_BC.sh
+			BiasFieldComputed=true
+			${CARET7DIR}/wb_command -cifti-separate ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.BiasField.native.dscalar.nii COLUMN \
+				-metric CORTEX_LEFT ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.L.BiasField.native.func.gii \
+				-metric CORTEX_RIGHT ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.R.BiasField.native.func.gii
+		else
+			# bias field in native space is already generated
+			# BC the other types of given myelin maps
+			${CARET7DIR}/wb_command -cifti-math "Var - Bias" ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.${MyelinMap}_BC.native.dscalar.nii -var Var ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.${MyelinMap}.native.dscalar.nii -var Bias ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.BiasField.native.dscalar.nii
+		fi
+		${CARET7DIR}/wb_command -cifti-separate ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.${MyelinMap}_BC.native.dscalar.nii COLUMN \
+			-metric CORTEX_LEFT ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.L.${MyelinMap}_BC.native.func.gii \
+			-metric CORTEX_RIGHT ${AtlasSpaceFolder}/${NativeFolder}/${Subject}.R.${MyelinMap}_BC.native.func.gii
```

Comment:
This final command of this loop can be moved to the start of the next loop, and combined with the above comments, that means this loop doesn't need to exist anymore.